### PR TITLE
Raise StormRuntimeError for invalid Storm funccalls (SYN-5261)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/ast.py [hunk 1]
```diff
@@ -2866,6 +2866,9 @@ class FuncCall(Value):
     async def compute(self, runt, path):
 
         func = await self.kids[0].compute(runt, path)
+        if not callable(func):
+            raise s_exc.StormRuntimeError(mesg=f"'{type(func).__name__}' object is not callable.")
```
Comment (vEpiphyte) [inline synapse/lib/ast.py:2870]: If we have the name of variable here that could be useful to include.

Comment (invisig0th) [inline synapse/lib/ast.py:2870]: Would we want to try to map these to storm types rather than allowing python types to slip through?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/ast.py (hunk 1 above)
Comment (Cisphyx) [reply synapse/lib/ast.py:2870]: We don't always, but after the AST changes are merged we would have the text to include here which may be better

Comment (Cisphyx) [reply synapse/lib/ast.py]: updated to use `totype`

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**